Cookbook/Technical-Documents/TaskReviewGuidelines.md

```diff
@@ -0,0 +1,60 @@
+
+# Technical test reviewer guidelines
+
+This document describes the guidelines for reviewing technical task assignments that candidates work on during the recruitment process.
+Every question is attributed with **S/M/J** or a combination of those, defining what level it relates to.
+
+- **S** (expected of a senior-level task)
+- **M** (expected of a mid-level task)
+- **J** (expected of a junior-level task)
+
+## General:
+
+### Basics
+- Does it have clear instructions on how to run the project? (SMJ)
+- Are there any compile time issues? (SMJ)
+- Does it meet functional requirements? (SMJ)
+- Are there any run time issues? (SM)
+- Does it have descriptive README? (SM)
+
+### Good practices
+
+- Is it testable? (S, some for M)
+- Does it conform to the SOLID principles? (S, some for M)
+- Does it use dependency injection? (S, some for M)
+- Does it use descriptive naming? (SM)
+- Is the code clean? (no dead code) (SM)
+- Does it use encapsulation? (SM)
+- Are there any warnings in the codebase? (SM)
+- Are modules loosely coupled? (S)
+- Does it use higher order functions? (bonus point)
+- Side effects pushed to the edge? (bonus point)
+- Is it modularized? (bonus point)
+
+## Architecture:
+
+### Networking
+- Does it use `Codable`? (SMJ)
+- Is it well abstracted? (S, some for M)
+- Is it open for extensions? (like authorization, different routes and HTTP methods) (SM)
+- Are errors propagated? (SM)
+- Are requests cancelable? (S)
+
+### Persistence
+- Is it well abstracted? (it does not leak implementation details to other layers) (S)
+
+### UI-Business logic
+- Does `UIViewController` has only its clear responsibilities? (SM, some for J)
+- Is the navigation abstracted? (SM)
+- Is business logic state management done within UI state/actions? (S)
```

Comment:
I would make it as a bonus point because, someone would do that probably only if the do some kind of unidirectional data flow